src/data/polynomial/semiring_division.lean

```diff
@@ -0,0 +1,586 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Johannes Hölzl, Scott Morrison, Jens Wagemaker
+-/
+import data.polynomial.monic
+import ring_theory.euclidean_domain
+import ring_theory.multiplicity
+
+/-!
+# Theory of univariate polynomials
+
+
+-/
```

Comment:
We need a better filename here. It's actually more general than about just `[semiring]`.

Perhaps just `polynomial/div.lean`?